File: magnets.js
// http://codeforces.com/contest/344/problem/A

var magnets = (n, arr) => {
  var temp = arr[0]
  var ans = 0

  for (var i = 1;i < n;i++) {
    if (arr[i][1] == temp[1]) {
      ans++
    }
    temp = arr[i]
  }

  console.log(ans)
  return ans
}

magnets(6, ['10', '10', '10', '01', '10', '10']) // 3
magnets(4, ['01', '01', '10', '10']) // 2

File: magnets.cpp
#include<iostream>
using namespace std;

int main() {
    int n;
    cin >> n;
    string arr[n];

    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    char check = arr[0][1];
    int ans = 1;

    for (int i = 1; i < n; i++) {
        if (arr[i][1] != check) {
            ans++;
            check = arr[i][1];
        }
    }
    cout << ans;
    return 0;
}